src/nvim/diff.c

```diff
@@ -2647,6 +2878,97 @@ bool diff_find_change(win_T *wp, linenr_T lnum, int *startp, int *endp)
   bool added = true;
 
   linenr_T off = lnum - dp->df_lnum[idx];
+  if (chardiff()) {
```

Comment:
this is another big block that probably can be extracted to a function.

Reply:
moved to seperate function